src/Centreon/Infrastructure/ServiceConfiguration/ServiceConfigurationRepositoryRDB.php

```diff
@@ -237,29 +253,48 @@ public function findOnDemandServiceMacros(int $serviceId, bool $isUsingInheritan
      */
     public function findCommandLine(int $serviceId): ?string
     {
+        /*
+         * CTE recurse request next release:
+         *   WITH RECURSIVE inherite AS (
+         *     SELECT service_id, service_template_model_stm_id, command_command_id
+         *     FROM `:db`.service
+         *     WHERE service_id = :service_id
+         *     UNION
+         *     SELECT service.command_command_id, service.service_template_model_stm_id, service.command_command_id
+         *     FROM `:db`.service
+         *     INNER JOIN inherite
+         *         ON inherite.service_template_model_stm_id = service.service_id
+         *         AND inherite.command_command_id IS NULL
+         *   )
+         *   SELECT command.command_line
+         *   FROM inherite
+         *   INNER JOIN `:db`.command
+         *       ON command.command_id = inherite.command_command_id
+         */
         $request = $this->translateDbName(
-            'WITH RECURSIVE inherite AS (
-            SELECT service_id, service_template_model_stm_id, command_command_id
-            FROM `:db`.service
-            WHERE service_id = :service_id
-            UNION
-            SELECT service.command_command_id, service.service_template_model_stm_id, service.command_command_id
-            FROM `:db`.service
-            INNER JOIN inherite
-                ON inherite.service_template_model_stm_id = service.service_id
-                AND inherite.command_command_id IS NULL
-            )
-            SELECT command.command_line
-            FROM inherite
-            INNER JOIN `:db`.command
-                ON command.command_id = inherite.command_command_id'
+            'SELECT
+                srv.service_id AS service_id, service_template_model_stm_id,
+                command.command_line
+             FROM `:db`.service srv
+                LEFT JOIN `:db`.command ON srv.command_command_id = command.command_id
+             WHERE srv.service_id = :service_id LIMIT 1'
         );
         $statement = $this->db->prepare($request);
-        $statement->bindValue(':service_id', $serviceId, \PDO::PARAM_INT);
-        $statement->execute();
 
-        if (($record = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
-            return (string)$record['command_line'];
+        $serviceMacros = [];
+        $loop = [];
+        while (!is_null($serviceId)) {
+            if (isset($loop[$serviceId])) {
+                break;
+            }
+            $loop[$serviceId] = 1;
+            $statement->bindValue(':service_id', $serviceId, \PDO::PARAM_INT);
+            $statement->execute();
+            $record = $statement->fetch(\PDO::FETCH_ASSOC);
+            if (!is_null($record['command_line'])) {
+                return (string)$record['command_line'];
+            }
+            $serviceId = $record['service_template_model_stm_id'];
```

Comment:
`$serviceId` seems not used
Missing recursive call to `$this->findCommandLine` I guess

Reply:
it's used in the while and the request ;)